app/main/download_data.py

```diff
@@ -238,3 +240,19 @@ def process_api_response(query_params: dict) -> tuple:
         return abort(500), f"Unknown content type: {content_type}"
 
     return content_type, file_content
+
+
+def process_async_download(query_params: dict) -> int:
+    """Calls data-store for a file download request.
+
+    :param query_params: Query parameters for the API request.
+    :return: The status code of the API response.
+    """
+    request_url = (
+        Config.DATA_STORE_API_HOST
+        + "/trigger_async_download"
+        + ("?" + urlencode(query_params, doseq=True) if query_params else "")
+    )
+    response = requests.post(request_url)
+
+    return response.status_code
```

Comment:
As this is a POST request, we should pass the data in the body of the request, not as query params.

```suggestion
def process_async_download(data: dict) -> int:
    request_url = Config.DATA_STORE_API_HOST + "/trigger_async_download"
    response = requests.post(request_url, json=data)

    return response.status_code
```

Reply:
Added parameters to request body.